pdfminer/converter.py

```diff
@@ -77,8 +75,18 @@ def render_image(self, name, stream):
     def paint_path(self, gstate, stroke, fill, evenodd, path):
         """Paint paths described in section 4.4 of the PDF reference manual"""
         shape = ''.join(x[0] for x in path)
+
+        # if path contains multiple subpaths, split them up and reprocess
+        if shape.count("m") > 1:
+            m_indices = [i for i, x in enumerate(shape) if x == "m"]
+            m_indices_zipped = zip(m_indices, m_indices[1:] + [None])
+            subpaths = [path[a:b] for a, b in m_indices_zipped]
+            for sp in subpaths:
+                self.paint_path(gstate, stroke, fill, evenodd, sp)
```

Comment:
Can you try to make this code a bit shorter? E.g. with something along the lines of:

```python
>>> for m in re.finditer(r'ml+h', 'mlllhmlllh'):
...     print(m.start(0), m.end(0))
(0, 5)
(5, 10)
```

Reply:
Clever! Adjusted in latest push. Opted for `r'm[^m]+'` as the pattern, to catch subpaths that might not exclusively use `l` (e.g., `c` for cubic Bezier paths). Not sure exactly how those are parsed in pdfminer, though; let me know if you foresee problems with the pattern.

Reply:
This regex pattern looks good! :)